src/main/scala/pl/project13/kaze/KazeClass.scala

```diff
@@ -97,26 +163,111 @@ object KazeClass {
+      val wrapper = Map("int" -> "Int.box", "long" -> "Long.box")
 
-      // companion object
       indent = "  "
-      sb.append(s"object $clazzName {\n")
-      sb.append(s"${indent}/** Scala API */\n")
-      sb.append(s"${indent}def apply() = new $clazzName()\n")
-      sb.append(s"${indent}/** Java API */\n")
-      sb.append(s"${indent}def getInstance() = apply()\n")
-      sb.append("}\n")
+      sb.append(s"${indent}override def hashCode(): Int =\n")
+      sb.append(s"${indent}  java.util.Objects.hash(\n")
+      indent += "    "
+      for {
+        m <- fields
+        mName = m.getName
+      }
+        if (box.contains(m.getType.getName)) {
+          sb.append(s"${indent}${wrapper(m.getType.getName)}($mName),\n")
+        } else sb.append(s"${indent}$mName,\n")
+      sb.delete(sb.length - 2, sb.length)
+      sb.append(")\n")
+    }
 
-      sb.result()
+    private def renderObjectApply(sb: StringBuilder): Unit = {
+      // Scala: all fields apply method
+      appendScalaApi(sb, indent)
+      sb.append(s"${indent}def apply(\n")
+      fields.foreach { m =>
+        val mName = m.getName
+        val mType = theType(m)
+        sb.append(s"${indent}  $mName: $mType,\n")
+      }
+      sb.delete(sb.length - 2, sb.length)
+      sb.append("\n")
+      sb.append(s"$indent): $clazzName = new $clazzName(\n")
+      fields.foreach { m =>
+        val mName = m.getName
+        val mType = theType(m)
+        sb.append(s"${indent}  $mName,\n")
+      }
+      sb.delete(sb.length - 2, sb.length)
+      sb.append("\n")
+      sb.append(s"${indent})\n")
+      sb.append("\n")
     }
 
+    private def renderObjectCreate(sb: StringBuilder): Unit = {
+      // Java: all fields create method
+      appendJavaApi(sb, indent)
+      sb.append(s"${indent}def create(\n")
+      fields.foreach { m =>
+        val mName = m.getName
+        val mType = theType(m)
+        val jType = if (mType.startsWith("Option[")) {
+          mType.replace("Option", "java.util.Optional")
+        } else if (mType.startsWith("List[")) {
+          mType.replace("List", "java.util.List")
+        } else if (mType == "scala.concurrent.duration.FiniteDuration")
+          "java.time.Duration"
+        else mType
+
+        sb.append(s"${indent}  $mName: $jType,\n")
+      }
+      sb.delete(sb.length - 2, sb.length)
+      sb.append("\n")
+      sb.append(s"$indent): $clazzName = new $clazzName(\n")
+      fields.foreach { m =>
+        val mName = m.getName
+        val mType = theType(m)
+        if (mType.startsWith("List[")) {
+          sb.append(s"${indent}  $mName.asScala.toList,\n")
+        } else if (mType.startsWith("Option[") ||
+          mType == "scala.concurrent.duration.FiniteDuration") {
+          sb.append(s"${indent}  $mName.asScala,\n")
+        } else
+          sb.append(s"${indent}  $mName,\n")
+      }
+      sb.delete(sb.length - 2, sb.length)
+      sb.append("\n")
+      sb.append(s"${indent})\n")
+    }
```

Comment:
Doesn't these "all fields" methods defeat the purpose of making easier to evolve? the intention was to use some kind of factory method that creates an empty/default instance, and then amend it using the `with` methods. New `with` methods can be added without trouble.

Reply:
See my comment above.